src/libAtomVM/globalcontext.c

```diff
@@ -552,6 +552,23 @@ int globalcontext_get_registered_process(GlobalContext *glb, int atom_index)
     return 0;
 }
 
+term globalcontext_get_registered_name_process(GlobalContext *glb, int local_process_id)
+{
+    struct ListHead *registered_processes_list = synclist_wrlock(&glb->registered_processes);
+    struct ListHead *item;
+    struct ListHead *tmp;
+    MUTABLE_LIST_FOR_EACH (item, tmp, registered_processes_list) {
+        struct RegisteredProcess *registered_process = GET_LIST_ENTRY(item, struct RegisteredProcess, registered_processes_list_head);
+        if (registered_process->local_process_id == local_process_id) {
+            int result = registered_process->atom_index;
+            synclist_unlock(&glb->registered_processes);
+            return term_from_atom_index(result);
+        }
+    }
+    synclist_unlock(&glb->registered_processes);
+    return term_nil();
```

Comment:
My opinion: on the long term I find confusing using term_nil() as special value meaning nil, since it means `[]`.
I rather use `term_invalid_term()`. But anyway, what do you think?

Reply:
no strong opinions. Let me know if any changes are needed.

personally I see [] more as a nil value than an invalid value.

but note this is special case:
https://github.com/bjorng/otp/blob/250a9b6c3e9ae20a59cf48856eb069c113bf1d5f/erts/preloaded/src/erlang.erl#L7818

```
..the process has no registered name. In this case, `[]` is returned. This strange behavior is
because of historical reasons, and is kept for backward compatibility.
```

having `term_empty_list()`  would be perfect. but honestly I know little about the specifics.

Reply:
I would use different approaches between the internal API aimed to write C code, and the API exposed from NIFs.
`term_invalid_term()` is good for the first case, since it is guaranteed to be always invalid, while `[]' (that is `term_nil`) might be valid or invalid according to the situation.